### PR TITLE
[WIP] Switch configs to be explicitly enabled

### PR DESCRIPTION
**Pull request by danme**
Changes as a result of https://github.com/ck3d/nix2nvimrc/pull/1

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -15,11 +15,17 @@
...
+        ((builtins.attrValues nix2nvimrcModules)
+          ++ (builtins.attrValues nix2nvimrcConfigs));
+      enableModules = map (i: i.enabler)
+        (builtins.attrValues nix2nvimrcConfigs);
     in
     {
       lib = import ./lib.nix;
 
-      inherit overlays nix2nvimrcModules nix2nvimrcConfigs;
+      inherit overlays nvimModules;
```
Comment (danme) [inline flake.nix:28]: Maybe better to provide two modules, like `nixosModules.abc`?
One `nvimModules.default`, which imports nix2nvimrc and configs
and one `nvimModules.config`, which contains only the configs from ck3d?

File: lib.nix [hunk 2]
```diff
@@ -6,9 +6,12 @@ lib:
     in
     builtins.listToAttrs
       (map
-        (f: {
+        (f: rec {
           name = lib.removeSuffix suffix f;
-          value = dir + "/${f}";
+          value = {
+            file = dir + "/${f}";
+            enabler = { configs.${name}.enable = lib.mkDefault true; };
```
Comment (ck3d) [inline lib.nix:13]: nice aproach, but implicite dependency that file name and config name is the same. We have to find a way to check this implicite dependency.

File: flake.nix (hunk 1 above)
Comment (ck3d) [reply flake.nix:28]: nvimModules is now a list and not an attribute set.

your proposal is:
nvimModules = {
  modules = {
    wrapper
    ..
  }
  configs = {
    lspconfig
    ...
  }
}
I think that is what I did with nix2nvimrsModules and nix2nvimrcConfigs

**Comment by ck3d** [comment]
I found a better approach to support selection of configurations. See following commits:
https://github.com/ck3d/ck3d-nvim-configs/commit/9319fc8d01136949f848d96e5a3fb80df1a97a0e
https://github.com/ck3d/ck3d-nvim-configs/commit/4eef1be9853774745177d4b528234301c94cdf31

**Pull request closed without merging**